test/runnable/cppa.d

```diff
@@ -1264,6 +1264,39 @@ void test16536()
     version(OSX) assert(pass16536(123) == 123);
 }
 
+/****************************************/
+// 15589 - extern(C++) virtual destructors are not put in vtbl[]
+
+extern(C++)
+{
+    class A15589
+    {
+        extern(D) static int[] dtorSeq;
+        struct S
+        {
+            this(int x) { this.x = x; }
+            ~this() { dtorSeq ~= x; }
+            int x;
+        }
+        ~this() { dtorSeq ~= 10; }
+        S s1 = S(1);
+        S s2 = S(2);
+    }
+    class B15589 : A15589
+    {
+        ~this() { dtorSeq ~= 20; }
+        S s3 = S(3);
+    }
```

Comment:
Would be good to test whether C++ and D agree on the position of the dtor in the vtable. Can you add some virtual functions, matching declarations in cppb.cpp and call the dtor from there?